### PR TITLE
Improve agg v2 e2e tests

### PR DESCRIPTION
**Pull request by igor-aptos**
Previously we were only doing transactions from single account, so if they were executed speculatively, they would fail on SEQUENCE_NUMBER_TOO_NEW. modifying to use different account for each transaction, so all can happen in parallel.

adding all modes that currently work

### CODE REVIEW

File: aptos-move/e2e-move-tests/src/aggregator_v2.rs [hunk 1]
```diff
@@ -48,7 +40,24 @@ pub fn initialize(
     }
     let account = harness.new_account_at(AccountAddress::ONE);
     assert_success!(harness.publish_package_cache_building(&account, &path));
-    (harness, account)
+
+    let txn_accounts: Vec<Account> = (0..txns)
```
Comment (vusirikala) [inline aptos-move/e2e-move-tests/src/aggregator_v2.rs:44]: Are these sender accounts for each transcation?

File: aptos-move/e2e-move-tests/src/aggregator_v2.rs [hunk 2]
```diff
@@ -189,267 +144,285 @@ pub fn init(
...
+        name: &str,
+        agg_loc: &AggLocation,
+    ) -> SignedTransaction {
+        self.txn_index += 1;
+        self.harness.create_entry_function(
+            &self.txn_accounts[self.txn_index % self.txn_accounts.len()],
+            str::parse(name).unwrap(),
+            vec![agg_loc.element_type.get_type_tag()],
+            vec![
+                bcs::to_bytes(&agg_loc.address).unwrap(),
```
Comment (vusirikala) [inline aptos-move/e2e-move-tests/src/aggregator_v2.rs:159]: This argument is added additionally. What is this for?

**Review by vusirikala — APPROVED** [review]

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/aggregator_v2_test.move [hunk 3]
```diff
@@ -152,81 +151,71 @@ module 0x1::aggregator_v2_test {
         f(value)
     }
 
-    public entry fun new<Element: drop + copy + store>(account: &signer, use_type: u32, i: u64, limit: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
-        let addr = signer::address_of(account);
+    public entry fun new<Element: drop + copy + store>(_account: &signer, addr: address, use_type: u32, i: u64, limit: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
         insert<Aggregator<Element>>(addr, use_type, i, aggregator_v2::create_aggregator(limit));
```
Comment (vusirikala) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/aggregator_v2_test.move:155]: What is `use_type`?

File: aptos-move/e2e-move-tests/src/aggregator_v2.rs (hunk 1 above)
Comment (igor-aptos) [reply aptos-move/e2e-move-tests/src/aggregator_v2.rs:44]: yes

File: aptos-move/e2e-move-tests/src/aggregator_v2.rs (hunk 2 above)
Comment (igor-aptos) [reply aptos-move/e2e-move-tests/src/aggregator_v2.rs:159]: previously we were using signer address. now signer is someone else, and so explicitly pass the address of the resource to use

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/aggregator_v2_test.move (hunk 3 above)
Comment (igor-aptos) [reply aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/aggregator_v2_test.move:155]: whether to use AggregatorInResource, AggregatorInTable or AggregatorInResourceGroup

File: aptos-move/e2e-move-tests/src/aggregator_v2.rs [hunk 4]
```diff
@@ -90,83 +99,29 @@ impl ElementType {
...
-    account: &'a Account,
+pub struct AggLocation {
+    address: AccountAddress,
     element_type: ElementType,
     use_type: UseType,
     index: u64,
 }
 
-impl<'a> AggLocation<'a> {
+impl AggLocation {
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/aggregator_v2.rs:109]: nit: I would spell Aggregator instead of Agg, doesn't seem too wordy

File: aptos-move/e2e-move-tests/src/aggregator_v2.rs [hunk 5]
```diff
@@ -90,83 +99,29 @@ impl ElementType {
 }
 
 #[derive(Debug)]
-pub struct AggLocation<'a> {
-    account: &'a Account,
+pub struct AggLocation {
+    address: AccountAddress,
     element_type: ElementType,
     use_type: UseType,
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/aggregator_v2.rs:105]: I think it would be great to add a comment about why we need this (to identify in resource/group, etc.)

File: aptos-move/e2e-move-tests/src/aggregator_v2.rs [hunk 6]
```diff
@@ -189,267 +144,285 @@ pub fn init(
...
-        snap_loc.element_type.value_to_bcs(expected)
-    );
-    create_entry_agg_func_with_arg(
-        harness,
-        "0x1::aggregator_v2_test::check_snapshot",
-        snap_loc,
-        expected,
-    )
-}
+    fn create_entry_agg_func_with_arg(
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/aggregator_v2.rs:166]: Why it is better to have _no_arg, _arg and _two_args functions over passing a vector of arguments?

File: aptos-move/e2e-tests/src/executor.rs [hunk 7]
```diff
@@ -96,6 +96,14 @@ const POSTFIX: &str = "_should_error";
 /// Maps block number N to the index of the input and output transactions
 pub type TraceSeqMapping = (usize, Vec<usize>, Vec<usize>);
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
```
[the remark this answers is not in the document]
Comment (igor-aptos) [reply aptos-move/e2e-tests/src/executor.rs:99]: as this is test, I'll leave to address those in that PR, after this one lands

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs [hunk 8]
```diff
@@ -237,32 +249,38 @@ pub struct TestEnvConfig {
...
             block_split: BlockSplit::TxnPerBlock
         }),
-        // For now only test whole blocks with execution enabled.
-        // Just(TestEnvConfig {
-        //     executor_mode: DEFAULT_EXECUTOR_MODE,
-        //     aggregator_execution_enabled: true,
-        //     block_split: BlockSplit::Whole
-        // }),
+        // Sequential execution doesn't have exchanges, so we cannot use BothComparison, nor block split
+        // TODO join once sequential exchange lands.
```
[the remark this answers is not in the document]
Comment (igor-aptos) [reply aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs:267]: it is in the PR after this one https://github.com/aptos-labs/aptos-core/pull/10524

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs [hunk 9]
```diff
@@ -2,35 +2,33 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{
-    aggregator_v2::{
```
[the remark this answers is not in the document]
Comment (igor-aptos) [reply aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs:5]: yeah, would've been earlier, but given time, I don't want to delay this one for that.

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs [hunk 10]
```diff
@@ -157,6 +164,11 @@ pub fn run_block_in_parts(
             return;
         }
         let (errors, txns): (Vec<_>, Vec<_>) = txn_block.into_iter().unzip();
+        println!(
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs:167]: Is this intentional?

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs [hunk 11]
```diff
@@ -336,53 +355,53 @@ proptest! {
         is_2_collocated in any::<bool>(),
         is_3_collocated in any::<bool>(),
     ) {
-        println!("Testing test_aggregator_lifetime {:?}", test_env);
-        let (mut h, acc) = setup(test_env.executor_mode, test_env.aggregator_execution_enabled);
-        let acc_2 = h.new_account_with_key_pair();
-        let acc_3 = h.new_account_with_key_pair();
+        println!("Testing test_multiple_aggregators_and_collocation {:?}", test_env);
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs:358]: I think we have a logger we can use in FakeExecutor, but maybe printlns are ok

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs [hunk 12]
```diff
@@ -336,53 +355,53 @@ proptest! {
...
+            let (cur_acc, idx_2) = if is_2_collocated { idx_1 += 1; (h.account.address(), idx_1) } else { (acc_2.address(), 0)};
+            AggregatorLocation::new(*cur_acc, element_type, use_type, idx_2)
         };
         let agg_3_loc = {
-            let (cur_acc, idx_3) = if is_3_collocated { idx_1 += 1; (&acc, idx_1) } else { (&acc_3, 0)};
-            AggLocation::new(cur_acc, element_type, use_type, idx_3)
+            let (cur_acc, idx_3) = if is_3_collocated { idx_1 += 1; (h.account.address(), idx_1) } else { (acc_3.address(), 0)};
+            AggregatorLocation::new(*cur_acc, element_type, use_type, idx_3)
         };
         println!("agg_1_loc: {:?}", agg_1_loc);
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs:373]: These prints seem like they should be removed?

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs [hunk 13]
```diff
@@ -392,30 +411,32 @@ proptest! {
...
-        cases: 5,
-        result_cache: prop::test_runner::basic_result_cache,
+        cases: 200,
+        // result_cache: prop::test_runner::basic_result_cache,
         .. ProptestConfig::default()
     })]
 
     #[test]
     fn test_aggregator_underflow(test_env in arb_test_env(4)) {
+        // aptos_types::transaction::TransactionStatus::Keep(aptos_types::transaction::ExecutionStatus::MiscellaneousError(Some(SPECULATIVE_EXECUTION_ABORT_ERROR)))
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs:421]: We need this?

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs [hunk 14]
```diff
@@ -392,30 +411,32 @@ proptest! {
...
+
         println!("Testing test_aggregator_underflow {:?}", test_env);
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs:423]: Same comments as for other prints

**Review by georgemitenkov — APPROVED** [review]
